scripts/head_requests.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+from concurrent.futures import (
+    ThreadPoolExecutor,
+)
+import logging
+import sys
+import time
+
+from azul import (
+    require,
+)
+from azul.http import (
+    http_client,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+http = http_client(log)
+
+
+def parse_args(argv):
+    parser = argparse.ArgumentParser(description='Make HEAD requests at a given rate.')
+    parser.add_argument('--url',
+                        required=True,
+                        metavar='URL',
+                        help='The URL to request, e.g. https://<deployment>/index/summary.')
+    parser.add_argument('--rate',
+                        required=True,
+                        metavar='RATE',
+                        type=int,
+                        help='The request rate in req/5min.')
+    parser.add_argument('--time',
+                        metavar='TIME',
+                        type=int,
+                        default='300',
+                        help='The length of the test in seconds (default: 300)')
+    args = parser.parse_args(argv)
+    require(args.url.startswith('http'))
+    require(1 <= args.rate <= 3000,
+            'Request rate must be between 1 and 3000')
+    require(1 < args.time <= 3600,
+            'Request time must be between 1 and 3600')
+    return args
+
+
+def head_url(url: str) -> int:
+    response = http.request('HEAD', url)
+    return response.status
+
+
+def main(argv):
+    args = parse_args(argv)
+    sleep_delay = 5 * 60 / args.rate
+    fudge = .98 - (args.rate / 100000)
+    sleep_delay *= fudge
```

Comment:
What's this for?

Reply:
The actual request rate ends up being slightly less than the desired. This fudge factor was an attempt to tweak the rate, but it turned out to be not accurate enough so instead I've removed it and added a note in the description of the `--rate` argument of the script.